gtk-osx-setup.sh

```diff
@@ -220,6 +220,11 @@ if test -d "$XDG_CONFIG_HOME" ; then
     config_dir="$XDG_CONFIG_HOME"
 elif test -d "$HOME/.config" ; then
     config_dir="$HOME/.config"
+else
+    config_dir="$HOME/.config"
+    if test ! -d $config_dir; then
+        mkdir -p $config_dir
+    fi
```

Comment:
We don't want to create ```$HOME/.config``` if ```$HOME/.jhbuildrc``` exists because we want to overwrite the latter and then ```$HOME/.config``` would be empty. On the other hand if ```$HOME/.jhbuildrc``` *doesn't* exist and ```$XDG_CONFIG_DIR``` is set and doesn't exist then we should probably be creating ```$XDG_CONFIG_DIR``` and putting jhbuildrc there instead of in ```$HOME/.config```.
We also always want to install a new ```jhbuildrc``` whether it exists or not. It's what customizes ```jhbuild``` for MacOS and needs to be in sync with the modulesets.

Reply:
So how about:
```
if test -e $HOME/.jhbuildrc; then
   JHBUILDRC=$HOME/.jhbuildrc
else
    if test -n $XDG_CONFIG_HOME; then
       config_dir="$XDG_CONFIG_HOME"
    else
       config_dir="$HOME/.config"
    fi
    if test ! -d "$config_dir"; then
        mkdir -p "$config_dir"
    fi
    JHBUILDRC="$config_dir/jhbuildrc"
fi
curl -ks $BASEURL/jhbuildrc-gtk-osx -o "$JHBUILDRC"
```

Reply:
Well, "test -n" doesn't behave as expected (?!) so I replaced it, I add also a comment.
NB: is it expected that jhbuildrc is every times replaced? That isn't an issue for me as when executing the script I expect to get the last one, but if not for you should add a presence test before the curl (as it was in previous version).

```suggestion
if test -e $HOME/.jhbuildrc; then
   JHBUILDRC=$HOME/.jhbuildrc
else
    if test ! -z $XDG_CONFIG_HOME; then
       config_dir="$XDG_CONFIG_HOME"
    else
       config_dir="$HOME/.config"
    fi
    if test ! -d "$config_dir"; then
        mkdir -p "$config_dir"
    fi
    JHBUILDRC="$config_dir/jhbuildrc"
fi
echo "Installing jhbuild configuration at $JHBUILDRC"
curl -ks $BASEURL/jhbuildrc-gtk-osx -o "$JHBUILDRC"
```

Reply:
Yes, it's expected that jhbuildrc is replaced every time. 
```
if test -n $XDG_CONFIG_HOME;
``` 
doesn't work but
```
if test -n "$XDG_CONFIG_HOME";
```
does. Makes sense, if ```$XDG_CONFIG_HOME``` isn't defined then the test is of the semicolon, a non-zero-length string. 
It occurs to me that ```$HOME/.jhbuildrc``` needs to be quoted too, there might be whitespace in ```$HOME```.

Last, I guess it's possible that ```$config_dir``` could exist but be a file instead of a directory, in which case ```mkdir -p "$config_dir"``` would fail.

Reply:
`if test -n "$XDG_CONFIG_HOME"; `
is effectively working better ;-)
If config_dir is a file, mkdir will indeed fail and report an error that warn for a potential configuration issue.